### PR TITLE
[RDAPI-26813] Policystudio: Modify Connection/ConnectToUrl filter UI to include new timeout overrides

### PR DESCRIPTION
**Pull request by dsantosaxway**
Policystudio: Modify Connection/ConnectToUrl filter UI to include new timeout overrides

## Describe the changes

Define custom timeouts for 'Connection' and 'ConnectToURL' filters so that policies can override RemoteHost timeouts when connecting to a URL

## Deploy preview link

https://deploy-preview-2291--axway-open-docs.netlify.app/docs/apim_policydev/apigw_polref/routing_common/#timeouts-settings

## Checklist for contributors

Before submitting this PR, please make sure:

_This is simply a reminder of what we are going to look for before merging your change._

* [ ] You have read the [contribution guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/)
* [ ] You have signed the [Axway CLA](https://cla.axway.com/)
* [ ] You have verified the technical accuracy of your change
* [ ] You have verified that your change does not expose any sensitive information (passwords, keys, etc.)
* [ ] You have followed the [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)  (unless this is is a Netlify CMS contribution)
* [ ] You have verified that all status checks have passed

### CODE REVIEW

File: content/en/docs/apim_policydev/apigw_polref/routing_common.md [hunk 1]
```diff
@@ -209,6 +210,22 @@ This option defaults to disabled, causing the connection not to be closed until
 
 {{< alert title="Note" color="primary" >}}This option replaces the system property `<VMArg name="-DConnectToUrlFilter.removePreviousConnections=true"/>` that was added in 7.5.3 SP3 to enable releasing previously opened connections. If you were using this system property, you must select this option on each connection filter requiring this behavior, as the system property no longer exists.{{< /alert >}}
 
+#### Timeouts settings
+
+Specify the timeouts settings, causing the filter to override the settings specified for Remote hosts, see [Advanced settings](https://axway-open-docs.netlify.app/docs/apim_policydev/apigw_gw_instances/general_remote_hosts/#advanced-settings)
+
```
Comment (dshielaxway) [inline content/en/docs/apim_policydev/apigw_polref/routing_common.md:216]: We're not just overriding Remote Host timeouts. Remote Host timeouts will only be employed if a remote host has been configured for the host:port specified for the URL in this filter.

By default we're overriding the global-level timeouts (specified at Server Settings > General). If a remote Host is configured, then the timeouts specified for that remote host will trump the global timeouts.

So the new timeouts in this filter will either override global-level timeouts, or remote host timeouts, depending on how the API Gateway is configured.

Comment (dsantosaxway) [reply content/en/docs/apim_policydev/apigw_polref/routing_common.md:216]: Thanks for the feedback, Would you agree with the explanation below? Feel free to rearrange the text in your comment for better understanding.

_"Specify the timeouts settings, causing the filter to override by default the global-level timeouts specified at Server Settings > General, see [General settings in Policy Studio](/docs/apim_reference/general_settings/#general-settings).

If a remote host is configured, then the timeouts specified for that remote host will override the global-level timeouts, see [Advanced settings](https://axway-open-docs.netlify.app/docs/apim_policydev/apigw_gw_instances/general_remote_hosts/#advanced-settings). 

Remote Host timeouts will only be employed if a remote host has been configured for the host: port specified for the URL in this filter.

The new timeouts in this filter will either override global-level timeouts or remote host timeouts, depending on how the API Gateway is configured."_

Comment (dshielaxway) [reply content/en/docs/apim_policydev/apigw_polref/routing_common.md:216]: How about the following...

The Timeouts section allows a Policy Developer to configure specific timeouts for this filter. These timeouts will override the equivalent global-level timeouts (see General settings in Policy Studio), or remote host level timeouts (see remote Host Advanced settings) if a remote host that matches the host:port of the URL configured in this filter happens to be configured.

**Override timeouts** : Specifies whether or not timeouts are overridden by this filter. The default is disabled.

**Connection Timeout** : Defaults to 30000 milliseconds (30 seconds). Specifying a negative value means an infinite timeout. Specifying a value of zero causes the filter to return immediately, regardless of the state of the underlying socket/file descriptor.

**Active Timeout** : Defaults to 30000 milliseconds (30 seconds). Specifying a negative value means an infinite timeout. Specifying a value of zero causes the filter to return immediately, regardless of the state of the underlying socket/file descriptor.

**Transaction Timeout (ms)** : Defaults to 240,000 milliseconds (4 minutes). Specifying a **Transaction Timeout** of less than or equal to zero means an infinite timeout.

**Idle Timeout** : Defaults to 15000 milliseconds (15 seconds). If a value less than or equal to zero is specified, the connection will be purged (almost) immediately.

Comment (dsantosaxway) [reply content/en/docs/apim_policydev/apigw_polref/routing_common.md:216]: Done! Thanks @dshielaxway 

Is it ok If I resolve this conversation?

File: content/en/docs/apim_policydev/apigw_polref/routing_common.md [hunk 2]
```diff
@@ -209,6 +209,20 @@ This option defaults to disabled, causing the connection not to be closed until
 
 {{< alert title="Note" color="primary" >}}This option replaces the system property `<VMArg name="-DConnectToUrlFilter.removePreviousConnections=true"/>` that was added in 7.5.3 SP3 to enable releasing previously opened connections. If you were using this system property, you must select this option on each connection filter requiring this behavior, as the system property no longer exists.{{< /alert >}}
 
+#### Timeouts settings
+
+Configure custom timeouts for a filter. The custom timeouts will override the equivalent global-level timeouts (see, [General settings in Policy Studio](/docs/apim_reference/general_settings/#general-settings)) or the remote host level timeouts (see, [Remote Host Advanced settings](/docs/apim_policydev/apigw_gw_instances/general_remote_hosts/#advanced-settings)) if a remote host that matches the `host:port` of the URL configured in the filter is configured.
```
Comment (andreamussap) [inline content/en/docs/apim_policydev/apigw_polref/routing_common.md:214]: (if a remote host that matches the `host:port` of the URL configured in the filter is configured) -> This is confusing - "If a remote host... **configured in the filter is configured**"  ??

configure where else?

Comment (dshielaxway) [reply content/en/docs/apim_policydev/apigw_polref/routing_common.md:214]: Hmm, I agree, it is confusing. How about
"If a remote host has been configured that matches the host:port of the URL specified in the filter"

Remote hosts are configured as separate entities in Policy Studio; see https://axway-open-docs.netlify.app/docs/apim_policydev/apigw_gw_instances/general_remote_hosts/

Comment (andreamussap) [reply content/en/docs/apim_policydev/apigw_polref/routing_common.md:214]: Thanks @dshielaxway! It reads better but I'll make just a minor change to improve it even more ;P

**Pull request merged**